[Pytorch][AO] Update choose_qparams_per_token op to output correct shape for scales and zp

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136807

- also makes scales and zp dtype reconcile with meta impl as well as other
quantized ops representation of scales and zero point
- make sure qunatize_per_token's output_dtype is respected

There are a few places where we need to reconcile on scale and zero point dtype
but that will come later. This fixes are mainly being done to enable quantized
kv cache though ET stack

Differential Revision: [D62301840](https://our.internmc.facebook.com/intern/diff/D62301840/)

kimishpatel committed:
[Pytorch][AO] Update choose_qparams_per_token op to output correct shape for scales and zp

- also makes scales and zp dtype reconcile with meta impl as well as other
quantized ops representation of scales and zero point
- make sure qunatize_per_token's output_dtype is respected

There are a few places where we need to reconcile on scale and zero point dtype
but that will come later. This fixes are mainly being done to enable quantized
kv cache though ET stack

Differential Revision: [D62301840](https://our.internmc.facebook.com/intern/diff/D62301840/)

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/fx/_decomposed.py:
```diff
@@ -954,7 +954,7 @@ def dequantize_per_token(
        input (torch.Tensor): quantized Tensor (uint8, int8 etc.)
-       scales (float32 torch.Tensor): quantization parameter for per token affine quantization
+       scales (float64 torch.Tensor): quantization parameter for per token affine quantization
        zero_points (int32 torch.Tensor): quantization parameter for per token affine quantization
```
should zero_points be int64?

kimishpatel committed:
Update on "[Pytorch][AO] Update choose_qparams_per_token op to output correct shape for scales and zp"


- also makes scales and zp dtype reconcile with meta impl as well as other
quantized ops representation of scales and zero point
- make sure qunatize_per_token's output_dtype is respected

There are a few places where we need to reconcile on scale and zero point dtype
but that will come later. This fixes are mainly being done to enable quantized
kv cache though ET stack

Differential Revision: [D62301840](https://our.internmc.facebook.com/intern/diff/D62301840/)

[ghstack-poisoned]
